### PR TITLE
Scan: Wire up proposal vote

### LINKED ISSUE #2615: Testing proposal flow
- Show each state correctly (Deposit / Voting / Passed / Rejected)
- User can vote on the proposal
- Show the deposit amount and depositor details correctly
- Show the total vote amount and each votes amount correctly (Yes / No / NoWithVeto / Abstain)
- Show the breakdown votes table correctly

### PR DESCRIPTION
**Pull request by pzshine**
Fixed: #2615 

## Implementation details

- Wired up the data to proposal vote
  - Turnout rate on Proposal Homepage
  - Proposal Index page
    - Turnout rate on Turnout pie chart
    - VoteStat on result box.
- Fixed some typo
- Added optionInt to ID.re

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: scan/src/pages/AccountIndexPage.re [hunk 1]
```diff
@@ -80,7 +80,7 @@ let balanceDetail = (~title, ~description, ~amount, ~usdPrice, ~color, ~isCountu
         tooltipPlacement=Text.AlignBottomStart
       />
     </Col>
-    <Col size=0.6>
+    <Col size=2.0>
```
Comment (evilpeach) [inline scan/src/pages/AccountIndexPage.re:83]: Why we have to change this?

File: scan/src/pages/ProposalIndexPage.re [hunk 2]
```diff
@@ -92,10 +92,26 @@ module VoteButton = {
   };
 };
 
+type answer_t =
+  | Yes
+  | No
+  | NoWithVeto
+  | Abstain;
+
```
Comment (evilpeach) [inline scan/src/pages/ProposalIndexPage.re]: Why don't we declare it on `VoteSub`?

File: scan/src/pages/ProposalIndexPage.re [hunk 3]
```diff
@@ -142,144 +158,214 @@ let make = (~proposalID) => {
...
+                          switch (proposalSub) {
+                          | Data({name, status}) =>
+                            <div className={Styles.voteButton(status)}>
+                              <VoteButton proposalID proposalName=name />
+                            </div>
+                          | _ => <LoadingCensorBar width=90 height=26 />
+                          };
+                        }}
+                   </div>
+                   //TODO: will re-structure when the data is wired up.
```
Comment (evilpeach) [inline scan/src/pages/ProposalIndexPage.re]: remove?

File: scan/src/pages/ProposalIndexPage.re [hunk 4]
```diff
@@ -142,144 +158,214 @@ let make = (~proposalID) => {
...
+                          | Data((_, {total}, bondedToken)) =>
+                            let turnoutPercent =
+                              total /. (bondedToken |> Coin.getBandAmountFromCoin) *. 100.;
+                            <TurnoutChart percent=turnoutPercent />;
+                          | _ => React.null
+                          }}
+                         {isMobile
+                            ? <div>
+                                <Heading value="Total" size=Heading.H5 marginBottom=8 />
+                                //TODO: wire up later
```
Comment (evilpeach) [inline scan/src/pages/ProposalIndexPage.re]: remove?

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — CHANGES_REQUESTED** [review]
Let's check https://d3n-scan-pr-2621.onrender.com/proposal/25. Why the percentage of abstain choice is 0?

File: scan/src/pages/AccountIndexPage.re (hunk 1 above)
Comment (pzshine) [reply scan/src/pages/AccountIndexPage.re:83]: Fixed the wrapped text

File: scan/src/subscriptions/VoteSub.re [hunk 5]
```diff
@@ -99,3 +211,175 @@ let count = (proposalID, answer) => {
...
+    parsedData->Belt_Array.reduce(0., (a, {validatorPower}) => a +. validatorPower);
+
+  let delegatorPower =
+    delVotes##non_validator_vote_proposals_view
+    ->Belt_Array.reduce(0., (a, {power}) => a +. (power |> Coin.getBandAmountFromCoin));
+
+  let totalPower = validatorPower +. delegatorPower;
+
+  let totalYesPower =
+    parsedData->getValVote(Yes) +. delVotes##non_validator_vote_proposals_view->getDelVote(Yes);
```
Comment (evilpeach) [inline scan/src/subscriptions/VoteSub.re]: declare `delVotes##non_validator_vote_proposals_view` as variable

File: scan/src/subscriptions/VoteSub.re [hunk 6]
```diff
@@ -99,3 +211,175 @@ let count = (proposalID, answer) => {
...
+    parsedData->getValVote(NoWithVeto)
+    +. delVotes##non_validator_vote_proposals_view->getDelVote(NoWithVeto);
+  let totalAbstainPower =
+    parsedData->getValVote(Abstain)
+    +. delVotes##non_validator_vote_proposals_view->getDelVote(Abstain);
+
+  Sub.resolve({
+    proposalID,
+    totalYes: totalYesPower,
+    totalYesPercent: totalYesPower != 0. ? totalYesPower /. totalPower *. 100. : 0.,
```
Comment (evilpeach) [inline scan/src/subscriptions/VoteSub.re]: remove `totalYesPower` and friends

File: scan/src/subscriptions/VoteSub.re [hunk 7]
```diff
@@ -99,3 +211,175 @@ let count = (proposalID, answer) => {
...
+    totalNoWithVeto: totalNoWithVetoPower,
+    totalNoWithVetoPercent:
+      totalNoWithVetoPower != 0. ? totalNoWithVetoPower /. totalPower *. 100. : 0.,
+    totalAbstain: totalAbstainPower,
+    totalAbstainPercent: totalAbstainPower != 0. ? totalAbstainPower /. totalPower *. 100. : 0.,
+    total: totalPower,
+  });
+};
+
+let getVoteStat = () => {
```
Comment (evilpeach) [inline scan/src/subscriptions/VoteSub.re]: getVoteStats

File: scan/src/subscriptions/VoteSub.re [hunk 8]
```diff
@@ -65,6 +125,58 @@ module VoteCountConfig = [%graphql
...
+        power: amount @bsDecoder(fn: "GraphQLParser.coinExn")
+        proposalID: proposal_id @bsDecoder(fn: "ID.Proposal.fromIntExn")
+      }
+    }
+  |}
+];
+
+module ValidatorVoteConfig = [%graphql
+  {|
+    subscription ValidatorVoteByProposalID {
```
Comment (evilpeach) [inline scan/src/subscriptions/VoteSub.re:156]: ValidatorVotes

File: scan/src/subscriptions/VoteSub.re [hunk 9]
```diff
@@ -65,6 +125,58 @@ module VoteCountConfig = [%graphql
...
+        power: amount @bsDecoder(fn: "GraphQLParser.coinExn")
+        proposalID: proposal_id @bsDecoder(fn: "ID.Proposal.fromIntExn")
+      }
+    }
+  |}
+];
+
+module DelegatorVoteConfig = [%graphql
+  {|
+    subscription DelegatorVoteByProposalID {
```
Comment (evilpeach) [inline scan/src/subscriptions/VoteSub.re:169]: DelegatorVotes

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**